### PR TITLE
finish task 7

### PR DESCRIPTION
**Pull request by XingWei0920**
I just finished task 7

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -156,3 +156,34 @@ describe("/api/article/:article_id", () => {
...
+      .send(newComment)
+      .then((response) => {
+        
+          expect(response.body.comment.author).toBe('lurker');
+          expect(response.body.comment.body).toBe("This is an excellent article");
+          expect(response.body.comment.article_id).toBe(1);
+      });
+  });
+
+  test("GET:201 post a comment into the comments table with a invalid article_id", () => {
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js]: 400

File: endpoints.json [hunk 2]
```diff
@@ -25,5 +25,21 @@
         }
       ]
     }
+  },
+  "POST /api/articles/:article_id/comments": {
+    "description": "post a comment for an article",
```
Comment (nc-jonathan-mcguire) [inline endpoints.json]: any details on the post body here would be very helpful - how does someone know what they need to post?

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
Just a few bits to add:

- [ ] See comment on the endpoints.json
- [ ] TEST: Status 404: Not found, article_id does not exist E.g. POST /api/articles/9999/comments
- [ ] TEST: Status 400: Bad request, incomplete body - Request body incomplete/missing required field(s) of username or body.
- [ ] TEST: Status 404: Not found, username does not exist

File: controllers/server-controllers.js [hunk 3]
```diff
@@ -38,5 +41,24 @@ exports.getArticleById= (req, res, next) => {
     .catch(next)
 }
 
+exports.postCommentsByArticleId = (req, res, next) => {
+        const {article_id}=req.params
+        const newComment=req.body
+        if (article_id)
```
Comment (ghost) [inline controllers/server-controllers.js]: This is not optional, the article_id will come from the parameters.

File: controllers/server-controllers.js [hunk 4]
```diff
@@ -38,5 +41,24 @@ exports.getArticleById= (req, res, next) => {
...
+            return checkArticle_idExists(article_id)
+            .then (()=>{
+                return checkUsernameExists(newComment.author)
+            })
+            .then(()=>{
+                return addNewComment(newComment,article_id)})
+            .then((comment)=>{
+                res.status(201).send({comment});
+            })
+            .catch(next)
```
Comment (ghost) [inline controllers/server-controllers.js]: If you're going to use three promises here, do a promise all.

**Review by ghost — APPROVED** [review]
Looks like you've added the right tests. A nice refactor would be to take those three promises in your controller and run a promise.all.

File: controllers/server-controllers.js [hunk 5]
```diff
@@ -41,24 +41,36 @@ exports.getArticleById= (req, res, next) => {
...
+//             })
+//             .then(()=>{
+//                 return addNewComment(newComment,article_id)})
+//             .then((comment)=>{
+//                 res.status(201).send({comment});
+//             })
+//             .catch(next)
+//         }
+// }
+
```
Comment (nc-mick-fay) [inline controllers/server-controllers.js:62]: Remove any commented out code before submitting your pull requests please

File: controllers/server-controllers.js [hunk 6]
```diff
@@ -41,24 +41,36 @@ exports.getArticleById= (req, res, next) => {
...
+    const newComment=req.body
+    const promises=[checkArticle_idExists(article_id),checkUsernameExists(newComment.author),addNewComment(newComment,article_id)]
+        Promise.all(promises)
+        .then((result)=>{
+            comment=result[2]
+            res.status(201).send({comment});
+        })
+        .catch(next)   
+}
+
```
Comment (nc-mick-fay) [inline controllers/server-controllers.js:74]: Really nice refactor to use Promise.all()

**Review by nc-mick-fay — APPROVED** [review]
Great refactor! Remove your commented out code and you're ready to merge into main

**Pull request merged**